pkg/agent/agent.go

```diff
@@ -388,8 +388,7 @@ func (a *CommonAgent) SendHeartbeat() error {
 
 	jsonData, err := json.Marshal(payload)
 	if err != nil {
-		a.Logger().Infof("Error marshaling JSON: %s", err)
-		fmt.Println("Error marshaling JSON:", err)
+		a.Logger().Errorf("Error marshaling JSON: %s", err)
```

Comment:
Somewhat debatable as to whether this is necessary since we eventually will log the error later